components/camera/camera.go

```diff
@@ -152,6 +152,16 @@ func NewPropertiesError(cameraIdentifier string) error {
 	return errors.Errorf("failed to get properties from %s", cameraIdentifier)
 }
 
+// OverwriteCameraVideoSource overwrites the given camera's video source with the one provided.
+func OverwriteCameraVideoSource(source gostream.VideoSource, camera Camera) error {
+	if asVideoSrc, ok := camera.(*videoSource); ok {
+		asVideoSrc.videoSource = source
+		asVideoSrc.videoStream = gostream.NewEmbeddedVideoStream(source)
+		return nil
+	}
+	return errors.Errorf("cannot convert camera (type %T) to type (%T)", camera, (*videoSource)(nil))
+}
```

Comment:
If the camera.Reconfigure command works for what you need done, then you might not need this function.